libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -356,10 +374,14 @@ AP_GPS::update_instance(uint8_t instance)
     uint32_t tnow = AP_HAL::millis();
 
     // if we did not get a message, and the idle timer of 1.2 seconds
-    // has expired, re-initialise the GPS. This will cause GPS
-    // detection to run again
+    // has expired, re-initialise the GPS. This will be delayed until
+    // after the current time is greater then the response_time_ms
+    // that is used by the driver to indicate that a long action may
+    // be underway until this time, and the driver should bot be unloaded.
+    // Unloading the driver will cause GPS detection to run again
     if (!result) {
-        if (tnow - timing[instance].last_message_time_ms > 1200) {
+        if (tnow - timing[instance].last_message_time_ms > 1200 &&
+            tnow > state[instance].response_time_ms) {
```

Comment:
Shouldn't you "reset" the responde_tims_ms at line 396? Otherwise a following message could be waiting for a longer response time than it should right?